### PR TITLE
Add EKS-D 1-22 dev build support

### PR DESCRIPTION
**Pull request by vivek-koppuru**
*Issue #, if available:*

*Description of changes:*

Add support for building dev release of 1-22 to start testing prior to official release. We are not enabling Bottlerocket builds for now till we have those builds ready, so testing with Ubuntu for now.

This should replace the current WIP PR that is opened: #400 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/kubernetes-sigs/image-builder/Makefile [hunk 1]
```diff
@@ -34,8 +34,15 @@ PACKER_VAR_FILES=$(FULL_OUTPUT_DIR)/config/kubernetes.json $(PACKER_TYPE_VAR_FIL
 # Kubeadm hardcodes the version of the pause image it pulls, even tho containerd config
 # overrides which image is actually used
 # Setting up this mapping so we can build the additional image into the image
+# The versions are dependent on the tag set for the Kubernetes version we are consuming vs what is hardcoded in the
+# kubeadm constants for that specific version:
+# Example for v1.22.4 on why we rely on the pause tag from v1.22.5, which is the previous version that we support:
```
Comment (jaxesn) [inline projects/kubernetes-sigs/image-builder/Makefile:39]: I think you mean 1.21.5 here

File: build/lib/eksd_releases.sh [hunk 2]
```diff
@@ -38,7 +40,12 @@ function build::eksd_releases::load_release_yaml() {
 function build::eksd_releases::get_release_yaml_url() {
     local -r release_branch=$1
     local -r release_number=$(yq e ".${release_branch}.number" ${REPO_ROOT}/EKSD_LATEST_RELEASES)
-    local -r yaml_url="https://distro.eks.amazonaws.com/kubernetes-${release_branch}/kubernetes-${release_branch}-eks-${release_number}.yaml"
+    local -r dev=$(yq e ".${release_branch}.dev" ${REPO_ROOT}/EKSD_LATEST_RELEASES)
```
Comment (jaxesn) [inline build/lib/eksd_releases.sh:43]: What are you doing about the bundle pointing to the eks distro prod domain for the various artifacts?

Comment (vivek-koppuru) [reply build/lib/eksd_releases.sh:43]: We are planning on fixing it from the EKS-D side

File: projects/kubernetes-sigs/image-builder/Makefile (hunk 1 above)
Comment (vivek-koppuru) [reply projects/kubernetes-sigs/image-builder/Makefile]: Good catch

**Review by jaxesn — APPROVED** [review]
/lgtm

**Pull request merged**